### PR TITLE
Fix dtype mismatch error

### PR DESCRIPTION
**Pull request by ivy-lv11**
Fix dtype mismatch error when `load_in_low_bit='bf16'` 
CPU and GPU both: RuntimeError: expected m1 and m2 to have the same dtype, but got: float != c10::BFloat16 and details could be found in the issue https://github.com/analytics-zoo/nano/issues/1111

### CODE REVIEW

**Comment by hkvision** [comment]
@rnwang04 Please take a look at it?

File: python/llm/src/ipex_llm/transformers/models/llama.py [hunk 1]
```diff
@@ -1324,6 +1324,10 @@ def native_sdp(query, key, value, attention_mask,
 
     # at inference time, for memory considerations, may not need to upcast attention to fp32
     attn_weights = nn.functional.softmax(attn_weights, dim=-1)
+
+    if attn_weights.dtype != value.dtype:
+        attn_weights = attn_weights.to(value.dtype)
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/models/llama.py:1329]: Just do it unconditionally?

**Comment by rnwang04** [comment]
This PR can fix llama, but will other model meet similar issues ? 
Shall we add `torch_dtype=torch.bfloat16` for  `load_in_low_bit='bf16'` as we did for fp16 too ?

**Comment by ivy-lv11** [comment]
> This PR can fix llama, but will other model meet similar issues ? Shall we add `torch_dtype=torch.bfloat16` for `load_in_low_bit='bf16'` as we did for fp16 too ?

Yes

**Review by rnwang04 — APPROVED** [review]
LGTM

**Pull request merged**